File: BilDunya/src/main/java/com/bildunya/dto/CreateCommentRequest.java
package com.bildunya.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CreateCommentRequest {

    @NotBlank(message = "Text is required")
    private String text;

    private Long parentCommentId;

    private Boolean isAnonymous = false;
}

File: bildunya_frontend/lib/data/models/create_comment_request.dart
/// [com.bildunya.dto.CreateCommentRequest]
class CreateCommentRequest {
  const CreateCommentRequest({
    required this.text,
    this.parentCommentId,
    this.isAnonymous = false,
  });

  final String text;
  final int? parentCommentId;
  final bool isAnonymous;

  Map<String, dynamic> toJson() => {
        'text': text,
        if (parentCommentId != null) 'parentCommentId': parentCommentId,
        'isAnonymous': isAnonymous,
      };
}
